src/backend/drm/compositor/mod.rs

```diff
@@ -2357,6 +2362,17 @@ where
             }
         }
 
+        // If we only have a cursor update and vrr is enabled,
```

Comment:
I don't understand the comment here, what does update refer to here? This would mean that there is nothing shown except a cursor. Is this a realistic scenario?

Reply:
The case here being "nothing is being updated, but the cursor". Shouldn't we already have filtered out elements without any damage?

The main use-case here is a single fullscreen surface + a hardware cursor. Moving the cursor shouldn't dictate the frame rate in that case.

Reply:
What happens is that we build a new state and while we do this we compare with the previous one, including the commit id for direct scan-out. If nothing changed we still build the state, but skip the plane.

Composition on the primary plane might complicate things a bit, depending on how we handle the cursor update skip. We only know after calling into the damage tracked rendered if something has been drawn.

I haven't checked, but it should work to put the logic to after rendering to the primary plane and evaluate if only the cursor plane is not skipped, in which case we can override it to skip. This should automatically put our frame to "empty".

Though we either have to update the current frame to not loose the rendered cursor buffer and re-draw it every call to render_frame. Or just not skip if the cursor plane has been rendered to, which might be way easier imo.
This should be doable by comparing the buffer on the cursor plane with the previous frame.

Reply:
We only re-draw the cursor plane if absolutely necessary, so it should not happen often and already exclude movement alone. So maybe it would be okay to not skip in this case? Otherwise I fear it might be too easy to end in an unexpected cursor redraw-loop.